### ISSUE TITLE
Is it possible to get sharp.node statically linked with all native libs for webpack/AWS Lambda?

### ISSUE THREAD

**Issue by dvsekhvalnov** [lovell/sharp#1261]
Good day,

more looking to advice. Here is scenario we have:

 - AWS lambda deployment
 - all functions bundled with webpack
- sharp v0.20.3

First of all we don't have issues running sharp on AWS lambda, it's working fine after following official docs.

The issue is bundling with webpack because of native addon `sharp.node`.  We got partial success using https://github.com/ushu/node-addon-loader which copies addons files and rewrite `require(...)` to use relative path. (e.g. `require('build/sharp.node`).  

The problem is that `sharp.node` itself referencing modules in `../../vendor/lib/*`. If we manually copy `vendor/lib` is more or less working, but it's pretty much impossible to bundle automatically with webpack given a lot of native tooling will be involved like `ldd` or `otool`.

Is it something feasible to get statically compiled `sharp.node` version which is not referencing any `.so` modules?

Thank you in advance.

**Comment by lovell** [comment]
Hello, have you tried using the [externals](https://webpack.js.org/configuration/externals/) feature of Webpack?

**Comment by dvsekhvalnov** [comment]
Hi @lovell ,

sure, marking `sharp` as `externals` will and do work. But diminish overall idea of using webpack. Sharp with all its dependencies will not be processed, optimized and bundled by webpack. 

Let me rephrase initial question little bit, `sharp` is great library, would be awesome if it can be more webpack friendly. :) Unfortunately i don't have experience with node.js native addons and libvips as a c/c++ library in particular so can't figure out mysefl what can be effort or estimations or is it feasible at all to get statically compiled addon? 

As an alternate option, is it possible to configure addon with different linker path during install? (E.g. change `../../vendor/libs` to arbitrary relative path like `./native/libs` for instance).

**Comment by lovell** [comment]
You'll need to compile libvips and any desired dependencies as static libraries then modify the [libraries](https://github.com/lovell/sharp/blob/master/binding.gyp#L122) directive to reference the `.a` files before rebuilding `sharp.node`.

libvips is LGPL so static linking against it (and any other LGPL licensed dependencies) is likely to make your code LGPL too.

Good luck!

**Comment by lovell** [comment]
Hopefully this answered your question but please feel free to re-open if not.

**Comment by Bessonov** [comment]
@lovell I'm not very confident in the how to compile everything statically. Is there any way to provide something like a setting or flag to compile everything statically? I'm fighting with azure functions to get the right runtime.
